roles/custom/matrix-nginx-proxy/templates/nginx/conf.d/matrix-domain.conf.j2

```diff
@@ -390,6 +390,99 @@ server {
 		ssl_session_timeout {{ matrix_nginx_proxy_ssl_session_timeout }};
 	{% endif %}
 
+	{% if matrix_nginx_proxy_proxy_media_repo_enabled %}
+		# Redirect all media endpoints to the media-repo
+		location ^~ /_matrix/media {
```

Comment:
Looks like you're adding serving for `/_matrix/media` endpoints in the vhost which serves federation traffic.

Why is this necessary?

Looking at thte [Server-Server API specs](https://spec.matrix.org/v1.9/server-server-api/), there are no `/_matrix/media` endpoints described. I wonder if this is here by mistake.

The same nginx `location` blocks appear in the other vhost in the same file (above), which handles client-server traffic. Those seems fine, but I'm not sure why all these client-related APIs are being added to the federation vhost here.

Reply:
Adding the endpoint redirection for the federation vhost is done to the fix issue where clients from other homeservers can not download media from the MMR. If not added, download requests from clients from other homeservers seem to be routed to the matrix server/synapse reverse proxy companion service instead, which results in the media not being found and downloads failing.

Per documentation of the MMR nginx setup, it states that the `/_matrix/media` redirection directive should be added to the `server` block for both client and federation requests: https://docs.t2bot.io/matrix-media-repo/deployment/nginx.html.

As you mention however it does look like there is no federation endpoints for media endpoints, so perhaps there may be another way to fix this issue. I'll take another look at this after fixing Traefik proxy setup with these changes.

Reply:
can confirm that media federation is not working without changes

Reply:
Alright, this is ready for review now.

In regards to why `/_matrix/media` and related endpoints have to be added to the federation blocks, I think MMR adds the federation port when downloading remote media via federation: https://github.com/MatMaul/matrix-media-repo/blob/ae254aca90bd1074774e3b7b7fc2df24cce4416e/controllers/download_controller/download_resource_handler.go#L192C1-L199C62.